src/mqt/ionshuttler/linear/architecture.py

```
@@ -0,0 +1,236 @@
+        Args:
+            data: Architecture mapping.
+
+        Returns:
+            A validated architecture.
+
+        Raises:
+            ValueError: If the mapping has an invalid shape or values.
+        """
+        if not isinstance(data, dict):
+            msg = "architecture must be a JSON object"
+            raise ValueError(msg)  # ruff: ignore[type-check-without-type-error] - Input validation uses ValueError.
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)